pkg/runtime/field_export_reconciler.go

```diff
@@ -0,0 +1,674 @@
+		cm.Data = make(map[string]string, 1)
+	}
+	cm.Data[key] = sourceValue
+
+	ackrtlog.DebugFieldExport(r.log, desired, "patching target config map")
+	err = r.kc.Patch(ctx, cm, patch)
+	if err != nil {
+		return err
+	}
+	ackrtlog.InfoFieldExport(r.log, desired, "patched target config map")
+
+	return nil
+}
+
+// writeToSecret will patch an existing secret to add an exported field value.
+// By default the key will be "<namespace>.<name>" using values from the
+// exporter that created it.
+func (r *fieldExportReconciler) writeToSecret(
+	ctx context.Context,
+	sourceValue string,
+	desired *ackv1alpha1.FieldExport,
+) error {
+	// Construct the data key
+	key := fmt.Sprintf("%s.%s", desired.Namespace, desired.Name)
+
+	// Get the initial secret
+	nsn := types.NamespacedName{
+		Name: *desired.Spec.To.Name,
+	}
+	if desired.Spec.To.Namespace != nil {
+		nsn.Namespace = *desired.Spec.To.Namespace
+	} else {
+		nsn.Namespace = desired.Namespace
+	}
+
+	secret := &corev1.Secret{}
+	err := r.apiReader.Get(ctx, nsn, secret)
+	if err != nil {
+		return errors.Wrap(err, "unable to get existing secret")
+	}
+
+	// Update the field
+	patch := client.StrategicMergeFrom(secret.DeepCopy())
+	if secret.Data == nil {
+		secret.Data = make(map[string][]byte, 1)
+	}
+	secret.Data[key] = []byte(sourceValue)
+
+	ackrtlog.DebugFieldExport(r.log, desired, "patching target secret")
+	err = r.kc.Patch(ctx, secret, patch)
+	if err != nil {
+		return err
+	}
+	ackrtlog.InfoFieldExport(r.log, desired, "patched target secret")
+
+	return nil
+}
+
+func (r *fieldExportReconciler) FilterAllExports(
+	ctx context.Context,
+	gk metav1.GroupKind,
+	nsn types.NamespacedName,
+) ([]ackv1alpha1.FieldExport, error) {
+	listed := &ackv1alpha1.FieldExportList{}
+	opts := []client.ListOption{
+		client.InNamespace(nsn.Namespace),
+	}
+	if err := r.apiReader.List(ctx, listed, opts...); err != nil {
+		return []ackv1alpha1.FieldExport{}, err
+	}
+
+	exports := []ackv1alpha1.FieldExport{}
+	for _, export := range listed.Items {
+		// Ensure we are working with managed exports
+		if !k8sctrlutil.ContainsFinalizer(&export, fieldExportFinalizerString) {
+			continue
+		}
+
+		// Check the reference matches our source resource
+		if !strings.EqualFold(export.Spec.From.Resource.Kind, gk.Kind) ||
+			!strings.EqualFold(*export.Spec.From.Resource.Name, nsn.Name) {
+			continue
+		}
+
+		exports = append(exports, export)
+	}
+
+	return exports, nil
+}
+
+// onError will patch the FieldExport with the given error and return the
+// same error back
+func (r *fieldExportReconciler) onError(
+	ctx context.Context,
+	res *ackv1alpha1.FieldExport,
+	err error,
+) error {
+	r.patchRecoverableCondition(ctx, res, err)
+	return err
+}
+
+// onSuccess will patch the FieldExport with a synced condition and
+// return any errors that occurred while patching
+func (r *fieldExportReconciler) onSuccess(
+	ctx context.Context,
+	res *ackv1alpha1.FieldExport,
+) error {
+	r.clearConditions(ctx, res)
```

Comment:
* Should we use `ACK.ResourceSynced` condition for FieldExport resource as well or create a new Condition `ACK.FieldsExported` ?

Reply:
The reason why I didn't: I see conditions as actions taken by the controller on the given object, or the state of that object itself. The field export reconciler takes an action on the configmap, and simply uses the fieldexport object as the configuration. 

That being said, I'm happy to add this in a future revision if people want to see the last time it was synced, for instance.

Reply:
Yeah. For FieldExport resource, it will be useful to  
* easily find the last Synced time during debugging
* Knowing whether FieldExport is still in process of being reconciled if application fails to start

Reply:
Okay. I might add a follow-up PR with these changes, because I think we are already pretty close with this one.